utility/src/test/java/org/oppia/android/util/networking/DebugNetworkConnectionUtilTest.kt

```diff
@@ -0,0 +1,278 @@
+  }
+
+  @Test
+  fun testGetCurrentConnectionStatus_activeWifi_noForcedConnection_returnsWifi() {
+    setNetworkConnectionStatus(
+      status = ConnectivityManager.TYPE_WIFI,
+      networkState = NetworkInfo.State.CONNECTED
+    )
+    assertThat(debugNetworkConnectionUtil.getCurrentConnectionStatus()).isEqualTo(
+      ConnectionStatus.LOCAL
+    )
+  }
+
+  @Test
+  fun testGetCurrentConnectionStatus_activeWifi_forceCellularConnection_returnsCellular() {
+    setNetworkConnectionStatus(
+      status = ConnectivityManager.TYPE_WIFI,
+      networkState = NetworkInfo.State.DISCONNECTED
+    )
+    debugNetworkConnectionUtil.setCurrentConnectionStatus(ConnectionStatus.CELLULAR)
+    assertThat(debugNetworkConnectionUtil.getCurrentConnectionStatus()).isEqualTo(
+      ConnectionStatus.NONE
+    )
+  }
+
+  @Test
+  fun testGetCurrentConnectionStatus_activeWifi_forceNoConnection_returnsNone() {
+    setNetworkConnectionStatus(
+      status = ConnectivityManager.TYPE_WIFI,
+      networkState = NetworkInfo.State.DISCONNECTED
+    )
+    debugNetworkConnectionUtil.setCurrentConnectionStatus(ConnectionStatus.NONE)
+    assertThat(debugNetworkConnectionUtil.getCurrentConnectionStatus()).isEqualTo(
+      ConnectionStatus.NONE
+    )
+  }
+
+  @Test
+  fun testGetCurrentConnectionStatus_nonActiveWifi_noForcedConnection_returnsNone() {
+    setNetworkConnectionStatus(
+      status = ConnectivityManager.TYPE_WIFI,
+      networkState = NetworkInfo.State.DISCONNECTED
+    )
+    assertThat(debugNetworkConnectionUtil.getCurrentConnectionStatus()).isEqualTo(
+      ConnectionStatus.NONE
+    )
+  }
+
+  @Test
+  fun testGetCurrentConnectionStatus_nonActiveWifi_forceWifiConnection_returnsNone() {
+    setNetworkConnectionStatus(
+      status = ConnectivityManager.TYPE_WIFI,
+      networkState = NetworkInfo.State.DISCONNECTED
+    )
+    debugNetworkConnectionUtil.setCurrentConnectionStatus(ConnectionStatus.LOCAL)
+    assertThat(debugNetworkConnectionUtil.getCurrentConnectionStatus()).isEqualTo(
+      ConnectionStatus.NONE
+    )
+  }
+
+  @Test
+  fun testGetCurrentConnectionStatus_nonActiveWifi_forceCellularConnection_returnsNone() {
+    setNetworkConnectionStatus(
+      status = ConnectivityManager.TYPE_WIFI,
+      networkState = NetworkInfo.State.DISCONNECTED
+    )
+    debugNetworkConnectionUtil.setCurrentConnectionStatus(ConnectionStatus.CELLULAR)
+    assertThat(debugNetworkConnectionUtil.getCurrentConnectionStatus()).isEqualTo(
+      ConnectionStatus.NONE
+    )
+  }
+
+  @Test
+  fun testGetCurrentConnectionStatus_activeCellular_noForcedConnection_returnsCellular() {
+    setNetworkConnectionStatus(
+      status = ConnectivityManager.TYPE_MOBILE,
+      networkState = NetworkInfo.State.CONNECTED
+    )
+    assertThat(debugNetworkConnectionUtil.getCurrentConnectionStatus()).isEqualTo(
+      ConnectionStatus.CELLULAR
+    )
+  }
+
+  @Test
+  fun testGetCurrentConnectionStatus_activeCellular_forceWifiConnection_returnsWifi() {
```

Comment:
Per the code, isn't this inactiveCellular?

Reply:
Actually, the test case was incorrect here. I have corrected it. PTAL.